### PR TITLE
PIN-4561 BKE catalog-process Allow Eservice Update if has no descriptors but Technology is changed

### PR DESCRIPTION
**Pull request by nttdata-rtorsoli**

### CODE REVIEW

File: src/test/scala/it/pagopa/interop/catalogprocess/CatalogProcessSpec.scala [hunk 1]
```diff
@@ -4,27 +4,27 @@ import akka.http.scaladsl.model.StatusCodes
...
+  EServiceNotFound
+  // EServiceRiskAnalysisNotFound,
+  // AttributeNotFound,
+  // EServiceWithDescriptorsNotDeletable
 }
 import it.pagopa.interop.catalogprocess.model._
 import it.pagopa.interop.commons.cqrs.service.ReadModelService
 import it.pagopa.interop.commons.utils._
-import it.pagopa.interop.commons.utils.service.OffsetDateTimeSupplier
+// import it.pagopa.interop.commons.utils.service.OffsetDateTimeSupplier
```
Comment (beetlecrunch) [inline src/test/scala/it/pagopa/interop/catalogprocess/CatalogProcessSpec.scala:27]: 🧹

File: src/test/scala/it/pagopa/interop/catalogprocess/CatalogProcessSpec.scala
Comment (beetlecrunch) [inline src/test/scala/it/pagopa/interop/catalogprocess/CatalogProcessSpec.scala:1]: Many tests are commented.

File: src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala [hunk 2]
```diff
@@ -552,7 +552,7 @@ final case class ProcessApiServiceImpl(
   private def deleteInterfaceOnTechnologyUpdate(newTechnology: EServiceTechnology, catalogItem: CatalogItem)(implicit
     contexts: Seq[(String, String)]
   ): Future[Unit] = {
-    if (catalogItem.technology.toApi != newTechnology)
+    if (catalogItem.descriptors.nonEmpty && catalogItem.technology.toApi != newTechnology)
       for {
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala]: having an exception that should never be returned is a bit misleading.
You can find below the suggestion we were talking about

```
      val descriptorAndInterface = catalogItem.descriptors.find(_.state == Draft).flatMap(d => d.interface.map(i => (d, i))
      
      descriptorAndInterface.fold(Future.unit)(di => catalogManagementService.deleteEServiceDocument(catalogItem.id.toString, di._1.id.toString, di._2.id.toString))
```

Comment (nttdata-rtorsoli) [reply src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala]: Right

**Review by beetlecrunch — APPROVED** [review]

**Review by galales — APPROVED** [review]

**Pull request merged**